pkg/apis/build/v1beta1/buildrun_types.go

```diff
@@ -0,0 +1,394 @@
+	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+}
+
+// BuildRunSpec defines the desired state of BuildRun
+type BuildRunSpec struct {
+	// Build refers to an embedded build specification
+	//
+	// +optional
+	Build *ReferencedBuild `json:"build,omitempty"`
+
+	// ServiceAccount refers to the kubernetes serviceaccount
+	// which is used for resource control.
+	// Default serviceaccount will be set if it is empty
+	//
+	// +optional
+	ServiceAccount *ServiceAccount `json:"serviceAccount,omitempty"`
+
+	// Timeout defines the maximum run time of this BuildRun.
+	//
+	// +optional
+	// +kubebuilder:validation:Format=duration
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Params is a list of key/value that could be used
+	// to set strategy parameters
+	//
+	// +optional
+	ParamValues []ParamValue `json:"paramValues,omitempty"`
+
+	// Output refers to the location where the generated
+	// image would be pushed to. It will overwrite the output image in build spec
+	//
+	// +optional
+	Output *Image `json:"output,omitempty"`
+
+	// State is used for canceling a buildrun (and maybe more later on).
+	//
+	// +optional
+	State *BuildRunRequestedState `json:"state,omitempty"`
+
+	// Env contains additional environment variables that should be passed to the build container
+	//
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Contains information about retention params
+	// +optional
+	Retention *BuildRunRetention `json:"retention,omitempty"`
+
+	// Volumes contains volume Overrides of the BuildStrategy volumes in case those are allowed
+	// to be overridden. Must only contain volumes that exist in the corresponding BuildStrategy
+	// +optional
+	Volumes []BuildVolume `json:"volumes,omitempty"`
+}
+
+// BuildRunRequestedState defines the buildrun state the user can provide to override whatever is the current state.
+type BuildRunRequestedState string
+
+// BuildRunRequestedStatePtr returns a pointer to the passed BuildRunRequestedState.
+func BuildRunRequestedStatePtr(s BuildRunRequestedState) *BuildRunRequestedState {
+	return &s
+}
+
+const (
+	// BuildRunStateCancel indicates that the user wants to cancel the BuildRun,
+	// if not already canceled or terminated
+	BuildRunStateCancel = "BuildRunCanceled"
+
+	// BuildRunStatePodEvicted indicates that if the pods got evicted
+	// due to some reason. (Probably ran out of ephemeral storage)
+	BuildRunStatePodEvicted = "PodEvicted"
+)
+
+// SourceResult holds the results emitted from the different sources
+type SourceResult struct {
+	// Name is the name of source
+	Name string `json:"name"`
+
+	// Git holds the results emitted from from the
+	// step definition of a git source
+	//
+	// +optional
+	Git *GitSourceResult `json:"git,omitempty"`
+
+	// Bundle holds the results emitted from from the
+	// step definition of bundle source
+	//
+	// +optional
+	Bundle *BundleSourceResult `json:"bundle,omitempty"`
```

Comment:
We had not captured that in the workshop, but given we decided to rename this to `ociArtifact`, we should do that here as well:

```suggestion
	// OciArtifact holds the results emitted from source step of the ociArtifact type
	//
	// +optional
	OciArtifact *OciArtifactSourceResult `json:"ociArtifact,omitempty"`
```

Reply:
added the same